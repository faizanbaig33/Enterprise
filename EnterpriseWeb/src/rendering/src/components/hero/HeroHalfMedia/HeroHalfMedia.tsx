// Global
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { HeroHalfMediaTheme } from './HeroHalfMedia.theme';
import { getEnum, getHeadingLevel } from 'lib/utils';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';

import { Headline } from 'src/helpers/Headline';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { ButtonGroup } from 'src/helpers/ButtonGroup';
import { ButtonVariants } from 'src/helpers/Button';
import { Eyebrow } from 'src/helpers/Eyebrow';
import { MediaPrimary } from 'src/helpers/Media';
import { ComponentBackgroundVariants } from 'src/helpers/Component/Component';

export type ImagePosition = 'right' | 'left';
export type HeroHalfMediaProps = Feature.EnterpriseWeb.Components.Hero.HeroHalfMedia;

const HeroHalfMedia = (props: HeroHalfMediaProps) => {
  const imagePosition = getEnum<ButtonVariants>(props.fields?.imgPosition) ?? 'right';
  const ctaRightAlign = imagePosition === 'right';
  const style = getEnum<ComponentBackgroundVariants>(props.fields?.backgroundColor) ?? 'white';

  const { themeData } = useTheme(HeroHalfMediaTheme(ctaRightAlign, !!props.fields?.body?.value));

  return (
    <Component
      variant="full"
      gap="gap-x-0"
      padding=" "
      backgroundVariant={style}
      dataComponent="hero/herohalfmedia"
      {...props}
    >
      <div className={themeData.classes.contentClasses.imageContainerClass}>
        <MediaPrimary maxH={'w-full'} {...props} additionalDesktopClasses="h-full" />
      </div>
      <div className={themeData.classes.contentClasses.copyContainerClass}>
        <Eyebrow
          useTag={getHeadingLevel('h2', props.fields?.eyebrow)}
          classes={themeData.classes.eyebrowClass.eyebrowContainer}
          {...props}
        />
        <Headline
          useTag={getHeadingLevel('h1', props.fields?.headlineLevel)}
          classes={themeData.classes.firstHeadline.headlineContainer}
          {...props}
        />
        {props.fields?.body && (
          <BodyCopy classes={themeData.classes.contentClasses?.body} {...props} />
        )}
        {props.fields?.cta1Link?.value?.href && (
          <ButtonGroup classes={themeData.classes.buttonGroupClass} {...props} />
        )}
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<HeroHalfMediaProps>(HeroHalfMedia);
