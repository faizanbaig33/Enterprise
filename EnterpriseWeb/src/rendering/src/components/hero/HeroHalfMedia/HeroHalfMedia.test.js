// Globalimport { hasDataComponent, snapshot } from 'lib/jest/test-utils';// Localimport HeroHalfMedia from './HeroHalfMedia';import defaultData from './HeroHalfMedia.mock-data';import darkData from './HeroHalfMedia.mock-data-dark';import grayData from './HeroHalfMedia.mock-data-gray';import leftAlignData from './HeroHalfMedia.mock-data-left-align';import videoData from './HeroHalfMedia.mock-data-video';it('renders correctly', () => {  const component = snapshot(HeroHalfMedia, { componentProps: defaultData });  hasDataComponent(component, 'hero/herohalfmedia');});it('dark renders correctly', () => {  const component = snapshot(HeroHalfMedia, { componentProps: darkData });  hasDataComponent(component, 'hero/herohalfmedia');});it('left align renders correctly', () => {  const component = snapshot(HeroHalfMedia, { componentProps: leftAlignData });  hasDataComponent(component, 'hero/herohalfmedia');});it('gray renders correctly', () => {  const component = snapshot(HeroHalfMedia, { componentProps: grayData });  hasDataComponent(component, 'hero/herohalfmedia');});it('video renders correctly', () => {  const component = snapshot(HeroHalfMedia, { componentProps: videoData });  hasDataComponent(component, 'hero/herohalfmedia');});