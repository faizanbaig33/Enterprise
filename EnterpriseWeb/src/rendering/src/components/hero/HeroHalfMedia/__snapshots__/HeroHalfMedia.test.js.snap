// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dark renders correctly 1`] = `
<section
  data-component="hero/herohalfmedia"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-black bg-theme-bg"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12   gap-x-0"
    >
      <div
        class="col-span-12 ml:col-span-6 ml:content-right ml:items-right"
      >
        <div
          class="relative hidden md:block aspect-video w-full   h-full"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="relative block w-full md:hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
            />
            <noscript />
          </span>
        </div>
      </div>
      <div
        class="col-span-12 ml:col-span-6 ml:flex ml:flex-col ml:self-center p-m pt-s ml:pt-m ml:px-xl xl:px-xxl ml:-order-1"
      >
        <div
          class="text-theme-text text-sm-xxs text-xxs uppercase mb-s"
        >
          <h2>
            Eyebrow
          </h2>
        </div>
        <div
          class="items-top font-sans  text-theme-text text-sm-l ml:text-l font-heavy mb-2 last:mb-0"
        >
          <h1>
            This is a headline
          </h1>
        </div>
        <div
          class="text-theme-body text-body font-regular mb-m last:mb-0"
        >
          <div
            class="body-copy"
          >
            <div>
              <p>
                <span>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 flex flex-col ml:flex-row items-start mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            class="flex w-fit items-center whitespace-nowrap rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray ml:mr-3 ml:first:ml-0 ml:last:mr-0 mr-m"
            href="/Hero-Half-Media"
            role="link"
            tabindex="0"
            target=""
            title=""
          >
            Call to action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`gray renders correctly 1`] = `
<section
  data-component="hero/herohalfmedia"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-gray bg-light-gray"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12   gap-x-0"
    >
      <div
        class="col-span-12 ml:col-span-6 ml:content-right ml:items-right"
      >
        <div
          class="relative hidden md:block aspect-video w-full   h-full"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="relative block w-full md:hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
            />
            <noscript />
          </span>
        </div>
      </div>
      <div
        class="col-span-12 ml:col-span-6 ml:flex ml:flex-col ml:self-center p-m pt-s ml:pt-m ml:px-xl xl:px-xxl ml:-order-1"
      >
        <div
          class="text-theme-text text-sm-xxs text-xxs uppercase mb-s"
        >
          <h2>
            Eyebrow
          </h2>
        </div>
        <div
          class="items-top font-sans  text-theme-text text-sm-l ml:text-l font-heavy mb-2 last:mb-0"
        >
          <h1>
            This is a headline
          </h1>
        </div>
        <div
          class="text-theme-body text-body font-regular mb-m last:mb-0"
        >
          <div
            class="body-copy"
          >
            <div>
              <p>
                <span>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 flex flex-col ml:flex-row items-start mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            class="flex w-fit items-center whitespace-nowrap rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray ml:mr-3 ml:first:ml-0 ml:last:mr-0 mr-m"
            href="/Hero-Half-Media"
            role="link"
            tabindex="0"
            target=""
            title=""
          >
            Call to action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`left align renders correctly 1`] = `
<section
  data-component="hero/herohalfmedia"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12   gap-x-0"
    >
      <div
        class="col-span-12 ml:col-span-6 ml:content-right ml:items-right"
      >
        <div
          class="relative hidden md:block aspect-video w-full   h-full"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="relative block w-full md:hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
            />
            <noscript />
          </span>
        </div>
      </div>
      <div
        class="col-span-12 ml:col-span-6 ml:flex ml:flex-col ml:self-center p-m pt-s ml:pt-m ml:px-xl xl:px-xxl"
      >
        <div
          class="text-theme-text text-sm-xxs text-xxs uppercase mb-s"
        >
          <h2>
            Eyebrow
          </h2>
        </div>
        <div
          class="items-top font-sans  text-theme-text text-sm-l ml:text-l font-heavy mb-2 last:mb-0"
        >
          <h1>
            This is a headline
          </h1>
        </div>
        <div
          class="text-theme-body text-body font-regular mb-m last:mb-0"
        >
          <div
            class="body-copy"
          >
            <div>
              <p>
                <span>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 flex flex-col ml:flex-row items-start mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            class="flex w-fit items-center whitespace-nowrap rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray ml:mr-3 ml:first:ml-0 ml:last:mr-0 mr-m"
            href="/Hero-Half-Media"
            role="link"
            tabindex="0"
            target=""
            title=""
          >
            Call to action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly 1`] = `
<section
  data-component="hero/herohalfmedia"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12   gap-x-0"
    >
      <div
        class="col-span-12 ml:col-span-6 ml:content-right ml:items-right"
      >
        <div
          class="relative hidden md:block aspect-video w-full   h-full"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="relative block w-full md:hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
            />
            <noscript />
          </span>
        </div>
      </div>
      <div
        class="col-span-12 ml:col-span-6 ml:flex ml:flex-col ml:self-center p-m pt-s ml:pt-m ml:px-xl xl:px-xxl ml:-order-1"
      >
        <div
          class="text-theme-text text-sm-xxs text-xxs uppercase mb-s"
        >
          <h2>
            Eyebrow
          </h2>
        </div>
        <div
          class="items-top font-sans  text-theme-text text-sm-l ml:text-l font-heavy mb-2 last:mb-0"
        >
          <h1>
            This is a headline
          </h1>
        </div>
        <div
          class="text-theme-body text-body font-regular mb-m last:mb-0"
        >
          <div
            class="body-copy"
          >
            <div>
              <p>
                <span>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 flex flex-col ml:flex-row items-start mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            class="flex w-fit items-center whitespace-nowrap rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray ml:mr-3 ml:first:ml-0 ml:last:mr-0 mr-m"
            href="/Hero-Half-Media"
            role="link"
            tabindex="0"
            target=""
            title=""
          >
            Call to action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`video renders correctly 1`] = `
<section
  data-component="hero/herohalfmedia"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12   gap-x-0"
    >
      <div
        class="col-span-12 ml:col-span-6 ml:content-right ml:items-right"
      >
        <div
          style="position: relative; overflow: hidden; width: 100%; padding-top: 56.25%;"
        >
          <iframe
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowfullscreen=""
            id="YWkfIwORD-M"
            loading="lazy"
            name="ytwrapper"
            src="https://www.youtube.com/embed/YWkfIwORD-M?enablejsapi=1&loop=1&controls=1&autoplay=0&disablekb=0&modestbranding=0&mute=true&playlist=YWkfIwORD-M"
            style="position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; width: 100%; height: 100%;"
            title="Youtube Video"
          />
        </div>
      </div>
      <div
        class="col-span-12 ml:col-span-6 ml:flex ml:flex-col ml:self-center p-m pt-s ml:pt-m ml:px-xl xl:px-xxl ml:-order-1"
      >
        <div
          class="text-theme-text text-sm-xxs text-xxs uppercase mb-s"
        >
          <h2>
            Eyebrow
          </h2>
        </div>
        <div
          class="items-top font-sans  text-theme-text text-sm-l ml:text-l font-heavy mb-2 last:mb-0"
        >
          <h1>
            This is a headline
          </h1>
        </div>
        <div
          class="text-theme-body text-body font-regular mb-m last:mb-0"
        >
          <div
            class="body-copy"
          >
            <div>
              <p>
                <span>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 flex flex-col ml:flex-row items-start mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            class="flex w-fit items-center whitespace-nowrap rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray ml:mr-3 ml:first:ml-0 ml:last:mr-0 mr-m"
            href="/Hero-Half-Media"
            role="link"
            tabindex="0"
            target=""
            title=""
          >
            Call to action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
